### PR TITLE
Use newer Boost on EPEL

### PR DESCRIPTION
**Pull request by lzaoral**
This is a preparation for #105.

### CODE REVIEW

File: import-boost.sh [hunk 1]
```diff
@@ -16,4 +16,3 @@ import_from 1.74.0                                                      \
     boost_1_74_0/boost/nowide/replacement.hpp                           \
     boost_1_74_0/boost/nowide/utf/convert.hpp                           \
     boost_1_74_0/boost/nowide/utf/utf.hpp                               \
-    boost_1_74_0/boost/property_tree/
```
Comment (kdudka) [inline import-boost.sh:19]: We should also remove `boost_1_74_0/boost/property_tree` from our git tree.

Comment (lzaoral) [reply import-boost.sh:19]: Definitely, thank you! Addressed in b7cb17a.

**Comment by kdudka** [comment]
Thanks!  It works fine on RHEL-7 and RHEL-8 but for some reason it stopped working on Gentoo Linux:
```
% make
mkdir -p csdiff_build
cd csdiff_build && cmake \
        -D 'CMAKE_BUILD_TYPE=RelWithDebInfo' \
        -D SANITIZERS='OFF' ..
-- User-provided flags: 
CMake Error at /usr/lib64/cmake/Boost-1.81.0/BoostConfig.cmake:141 (find_package):
  Found package configuration file:

    /usr/lib64/cmake/boost_python-1.81.0/boost_python-config.cmake

  but it set boost_python_FOUND to FALSE so package "boost_python" is
  considered to be NOT FOUND.  Reason given by package:

  No suitable build variant has been found.

  The following variants have been tried and rejected:

  * libboost_python310.so.1.81.0 (3.10, Boost_PYTHON_VERSION=3.11)

Call Stack (most recent call first):
  /usr/lib64/cmake/Boost-1.81.0/BoostConfig.cmake:262 (boost_find_component)
  /usr/share/cmake/Modules/FindBoost.cmake:594 (find_package)
  src/CMakeLists.txt:103 (find_package)
  src/CMakeLists.txt:137 (build_pycsdiff)


-- Configuring incomplete, errors occurred!
See also "/home/kdudka/git/csdiff/csdiff_build/CMakeFiles/CMakeOutput.log".
make: *** [Makefile:32: all] Error 1
```

I was not able to decrypt the error message but the rejected library seems to be available in the default search path:
```
% file /usr/lib64/libboost_python310.so.1.81.0             
/usr/lib64/libboost_python310.so.1.81.0: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, stripped
```

Any idea how to make it work again?

**Comment by kdudka** [comment]
The problem seems to be that the script looks for `boost_python311` while only `boost_python310` is available.  But both `python` and `python3` are `3.10.x`:
```
% python3 --version
Python 3.10.9
% python --version 
Python 3.10.9
```

So why does cmake want to use Python 3.11?
```
-- Found Boost: /usr/lib64/cmake/Boost-1.81.0/BoostConfig.cmake (found version "1.81.0") found components: filesystem program_options regex system 
-- Found Python3: /usr/bin/python3.11 (found version "3.11.1") found components: Development Interpreter Development.Module Development.Embed 
```

**Comment by lzaoral** [comment]
Can you post a report with `-DBoost_DEBUG=ON` please?

**Comment by kdudka** [comment]
```
% make CMAKE='cmake -DBoost_DEBUG=ON'
removed 'src/lib/version.cc'
'src/lib/version.cc.tmp' -> 'src/lib/version.cc'
mkdir -p csdiff_build
cd csdiff_build && cmake -DBoost_DEBUG=ON \
        -D 'CMAKE_BUILD_TYPE=RelWithDebInfo' \
        -D SANITIZERS='OFF' ..
-- User-provided flags: 
-- Found Boost 1.81.0 at /usr/lib64/cmake/Boost-1.81.0
--   Requested configuration: QUIET REQUIRED COMPONENTS filesystem;program_options;regex;system
-- BoostConfig: find_package(boost_headers 1.81.0 EXACT CONFIG REQUIRED QUIET HINTS /usr/lib64/cmake)
-- Found boost_headers 1.81.0 at /usr/lib64/cmake/boost_headers-1.81.0
-- BoostConfig: find_package(boost_filesystem 1.81.0 EXACT CONFIG REQUIRED QUIET HINTS /usr/lib64/cmake)
-- Found boost_filesystem 1.81.0 at /usr/lib64/cmake/boost_filesystem-1.81.0
-- Boost toolset is gcc12 (GNU 12.2.1)
-- Scanning /usr/lib64/cmake/boost_filesystem-1.81.0/libboost_filesystem-variant*.cmake
--   Including /usr/lib64/cmake/boost_filesystem-1.81.0/libboost_filesystem-variant-shared.cmake
--   [x] libboost_filesystem.so.1.81.0
-- Adding boost_filesystem dependencies: atomic;headers
-- Found boost_atomic 1.81.0 at /usr/lib64/cmake/boost_atomic-1.81.0
-- Boost toolset is gcc12 (GNU 12.2.1)
-- Scanning /usr/lib64/cmake/boost_atomic-1.81.0/libboost_atomic-variant*.cmake
--   Including /usr/lib64/cmake/boost_atomic-1.81.0/libboost_atomic-variant-shared.cmake
--   [x] libboost_atomic.so.1.81.0
-- Adding boost_atomic dependencies: headers
-- BoostConfig: find_package(boost_program_options 1.81.0 EXACT CONFIG REQUIRED QUIET HINTS /usr/lib64/cmake)
-- Found boost_program_options 1.81.0 at /usr/lib64/cmake/boost_program_options-1.81.0
-- Boost toolset is gcc12 (GNU 12.2.1)
-- Scanning /usr/lib64/cmake/boost_program_options-1.81.0/libboost_program_options-variant*.cmake
--   Including /usr/lib64/cmake/boost_program_options-1.81.0/libboost_program_options-variant-shared.cmake
--   [x] libboost_program_options.so.1.81.0
-- Adding boost_program_options dependencies: headers
-- BoostConfig: find_package(boost_regex 1.81.0 EXACT CONFIG REQUIRED QUIET HINTS /usr/lib64/cmake)
-- Found boost_regex 1.81.0 at /usr/lib64/cmake/boost_regex-1.81.0
-- Boost toolset is gcc12 (GNU 12.2.1)
-- Scanning /usr/lib64/cmake/boost_regex-1.81.0/libboost_regex-variant*.cmake
--   Including /usr/lib64/cmake/boost_regex-1.81.0/libboost_regex-variant-shared.cmake
--   [x] libboost_regex.so.1.81.0
-- Adding boost_regex dependencies: headers
-- BoostConfig: find_package(boost_system 1.81.0 EXACT CONFIG REQUIRED QUIET HINTS /usr/lib64/cmake)
-- Found boost_system 1.81.0 at /usr/lib64/cmake/boost_system-1.81.0
-- Boost toolset is gcc12 (GNU 12.2.1)
-- Scanning /usr/lib64/cmake/boost_system-1.81.0/libboost_system-variant*.cmake
--   Including /usr/lib64/cmake/boost_system-1.81.0/libboost_system-variant-shared.cmake
--   [x] libboost_system.so.1.81.0
-- Adding boost_system dependencies: headers
-- Found Boost 1.81.0 at /usr/lib64/cmake/Boost-1.81.0
--   Requested configuration: QUIET REQUIRED COMPONENTS python311
-- BoostConfig: find_package(boost_headers 1.81.0 EXACT CONFIG REQUIRED QUIET HINTS /usr/lib64/cmake)
-- BoostConfig: find_package(boost_python 1.81.0 EXACT CONFIG REQUIRED QUIET HINTS /usr/lib64/cmake)
-- Found boost_python 1.81.0 at /usr/lib64/cmake/boost_python-1.81.0
-- Boost toolset is gcc12 (GNU 12.2.1)
-- Scanning /usr/lib64/cmake/boost_python-1.81.0/libboost_python-variant*.cmake
--   Including /usr/lib64/cmake/boost_python-1.81.0/libboost_python-variant-shared-py3.10.cmake
--   [ ] libboost_python310.so.1.81.0
CMake Error at /usr/lib64/cmake/Boost-1.81.0/BoostConfig.cmake:141 (find_package):
  Found package configuration file:

    /usr/lib64/cmake/boost_python-1.81.0/boost_python-config.cmake

  but it set boost_python_FOUND to FALSE so package "boost_python" is
  considered to be NOT FOUND.  Reason given by package:

  No suitable build variant has been found.

  The following variants have been tried and rejected:

  * libboost_python310.so.1.81.0 (3.10, Boost_PYTHON_VERSION=3.11)

Call Stack (most recent call first):
  /usr/lib64/cmake/Boost-1.81.0/BoostConfig.cmake:262 (boost_find_component)
  /usr/share/cmake/Modules/FindBoost.cmake:594 (find_package)
  src/CMakeLists.txt:103 (find_package)
  src/CMakeLists.txt:137 (build_pycsdiff)


-- Configuring incomplete, errors occurred!
See also "/home/kdudka/git/csdiff/csdiff_build/CMakeFiles/CMakeOutput.log".
make: *** [Makefile:32: all] Error 1
```

**Comment by lzaoral** [comment]
Also, this seems like a bug with CMake packaging on Gentoo.  I would also expect that CMake would look for `/usr/bin/python3` first.  The same behaviour was already in the original POR for Boost Python: https://github.com/csutils/csdiff/pull/110#issuecomment-1431270688

I had to look for a particular version of libboost_python because older releases on e.g. Ubuntu do not support lookup only using the major version.

**Review by kdudka — APPROVED** [review]
As pointed out by @lzaoral the build failure on Gentoo is a separate issue: https://bugs.gentoo.org/835799

**Comment by lzaoral** [comment]
Thanks for the review!

**Pull request merged**